internal/gatewayapi/contexts.go

```diff
@@ -275,27 +273,27 @@ func GetRouteParentContext(route RouteContext, forParentRef gwapiv1.ParentRefere
 	}
 
 	routeParentStatusIdx := -1
+	defaultNamespace := gwapiv1.Namespace(metav1.NamespaceDefault)
 	statusParents := rv.FieldByName("Status").FieldByName("Parents")
 	for i := 0; i < statusParents.Len(); i++ {
 		p := statusParents.Index(i).FieldByName("ParentRef").Interface().(gwapiv1.ParentReference)
-		if !isHTTPRoute {
-			p = UpgradeParentReference(p)
-			defaultNamespace := gwapiv1.Namespace(metav1.NamespaceDefault)
-			if forParentRef.Namespace == nil {
-				forParentRef.Namespace = &defaultNamespace
-			}
-			if p.Namespace == nil {
-				p.Namespace = &defaultNamespace
-			}
+		// For those non-v1 routes, their underlying type of `ParentReference` is v1 as well.
+		// So we can skip upgrading these routes for simplicity.
+		if forParentRef.Namespace == nil {
+			forParentRef.Namespace = &defaultNamespace
+		}
+		if p.Namespace == nil {
+			p.Namespace = &defaultNamespace
 		}
 		if reflect.DeepEqual(p, forParentRef) {
 			routeParentStatusIdx = i
 			break
 		}
 	}
+
 	if routeParentStatusIdx == -1 {
 		tmpPR := forParentRef
-		if !isHTTPRoute {
+		if !isV1Route {
```

Comment:
do we need this code now that ParentReference is v1 ?

Reply:
no need, removed